pandas/core/arrays/arrow/array.py

```diff
@@ -653,6 +653,15 @@ def factorize(
         use_na_sentinel: bool = True,
     ) -> tuple[np.ndarray, ExtensionArray]:
         null_encoding = "mask" if use_na_sentinel else "encode"
+
+        pa_type = self._data.type
```

Comment:
Similar to https://github.com/pandas-dev/pandas/pull/50688#discussion_r1068596332, could you see if going through `cast` is more performant here?

Reply:
```
import pandas as pd
import pyarrow as pa

parr = pa.array(range(10000), type=pa.duration("s"))
arr = pd.core.arrays.ArrowExtensionArray(parr)

%timeit arr.factorize()
377 µs ± 17.6 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)  # <- astype
335 µs ± 2.92 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)  # <- cast
```

Reply:
so ~10% more performant at a similar complexity cost

Reply:
if this is a blocker ill change on the affected PRs.  i still like this pattern marginally more than the alternative, but mainly i want to get the slow xfails out of my workflow

Reply:
I would prefer going through `cast` (and generally keep ops in pyarrow-land as much as possible). If you could adjust this in #50688 that'd be good.